syntax = "proto3";

package payment;

message User {
  string user_id = 1;
  string name = 2;
  float balance = 3;
}

message PaymentRequest {
  string sender_id = 1;
  string recipient_id = 2;
  float amount = 3;
}

message Transaction {
  string transaction_id = 1;
  string sender_id = 2;
  string recipient_id = 3;
  float amount = 4;
  string timestamp = 5;
}

message PaymentResponse {
  bool success = 1;
  string message = 2;
  Transaction transaction = 3;
}
